packages/pyroscope-flamegraph/src/convert/sandwichViewProfiles.ts

```diff
@@ -53,37 +53,40 @@ export const treeToFlamebearer = (tree: TreeNode): FlamebearerData => {
 
 const arrayToTree = (
   nodesArray: TreeNode[],
-  maxLvlNumber: number,
   total: number
 ): TreeNode => {
   const result = {};
   let nestedObj = result;
-  const emptyLvls = maxLvlNumber - nodesArray.length;
-
-  for (let i = 0; i < emptyLvls; i += 1) {
-    const emptyNode = {
-      name: '',
-      key: '',
-      total: [0],
-      self: [0],
-      children: [],
-      offset: total,
-    };
-
-    nestedObj.children = [emptyNode];
-    nestedObj = emptyNode;
-  }
 
   nodesArray.forEach(({ name, ...rest }) => {
     const nextNode = { name, ...rest, total: [total] };
-
     nestedObj.children = [nextNode];
     nestedObj = nextNode;
   });
 
   return result.children[0];
 };
 
+function dedupTree(node) {
```

Comment:
can you add some comments about whats happening here?

Reply:
It's a recursive function that takes a node as input and then for that node walks over all children and performs deduplication meaning that it finds children with the same name and when two children have the same name it unifies them into 1 child node.

It then recursively calls itself on each new child node.

Reply:
can we type `node`?